common/src/main/java/org/opensearch/ml/common/connector/HttpConnector.java

```diff
@@ -324,13 +324,39 @@ public <T> T createPayload(String action, Map<String, String> parameters) {
             payload = fillNullParameters(parameters, payload);
             StringSubstitutor substitutor = new StringSubstitutor(parameters, "${parameters.", "}");
             payload = substitutor.replace(payload);
-
-            if (!isJson(payload)) {
-                throw new IllegalArgumentException("Invalid payload: " + payload);
+            boolean isJson = isJson(payload);
+            if (!isJson) {
```

Comment:
I think we can get rid of line 327 and use isJson(payload) directly in line 328.

Reply:
sounds good!